packages/core/lib/debug/printer.js

```diff
@@ -209,10 +210,33 @@ class DebugPrinter {
       this.config.logger.log(DebugUtils.formatStack(step.stack));
       this.config.logger.log("");
     }
-    this.config.logger.log(
-      DebugUtils.formatInstruction(traceIndex + 1, totalSteps, instruction)
-    );
-    this.config.logger.log(DebugUtils.formatPC(step.pc));
+
+    // printout instructions
+    const previousInstructions = 3;
+    const upcomingInstructions = 3;
+    const currentIndex = instruction.index;
+    // printout 3 previous instructions
+    if (currentIndex - previousInstructions + 1 > 0) {
+      this.config.logger.log("...");
+    }
+    for (let i = currentIndex - previousInstructions; i < currentIndex; i++) {
+      if (i >= 0) {
```

Comment:
Sorry, one more comment -- similar to the `Math.min` thing, I would suggest replacing this `if` with a `Math.max` in the bound, so you don't have to use a `for/if` structure.

Reply:
> > By the way @dongmingh, I notice you seem to always rebase and force-push. This makes it harder to keep track of. I would suggest that more often you should simply add new commits and do a normal push. Obviously there are some times where that's not tenable, but when it is tenable, it keeps the history clearer.
> 
> I also had a hard time following the conversation over time because of the force pushes. We should explain this in our dev-handbook. I'll post a PR.

I just git push the PR with no force.  Thanks.